### PR TITLE
LVM-501: Implement soft deletion for students

### PR DESCRIPTION
**Pull request by zack**
https://cs5500-jira.ccs.neu.edu/browse/LVM-501

### CODE REVIEW

File: app/controllers/students_controller.rb [hunk 1]
```diff
@@ -6,15 +6,15 @@ class StudentsController < ApplicationController
   add_breadcrumb 'Home', :root_path
 
   def index
-    add_breadcrumb 'Students', students_path
```
Comment (zack) [inline app/controllers/students_controller.rb:9]: This wasn't serving a purpose.

**Review by xingyif — COMMENTED** [review]
The `delete student` button on the Welcome page is off.
screen shot 2017-04-17 at 14 42 42

**Comment by zack** [comment]
@xingyif it's because your screen is small. We need to update all of those buttons. We can do it in another PR.

**Review by xingyif — APPROVED** [review]

File: app/controllers/students_controller.rb [hunk 2]
```diff
@@ -85,6 +110,31 @@ def update_tags
     redirect_to tutor_path(@student)
   end
 
+  def reinstate
+    @student = Student.of(current_user).find(params[:id])
```
Comment (xingyif) [inline app/controllers/students_controller.rb:114]: this is already done by `:set_student`.
redundant?

File: app/controllers/students_controller.rb [hunk 3]
```diff
@@ -85,6 +110,31 @@ def update_tags
...
+    @student.update(
+      deleted_on: nil,
+      deleted_by: nil
+    )
+
+    redirect_to @student
+  end
+
+  def delete
+    @student = Student.of(current_user).find(params[:id])
```
Comment (xingyif) [inline app/controllers/students_controller.rb:124]: already done by `:set_student`

File: app/controllers/students_controller.rb [hunk 4]
```diff
@@ -85,6 +110,31 @@ def update_tags
...
+
+    if current_user.admin?
+      redirect_to @student
+    else
+      redirect_to students_path
+    end
+  end
+
   def destroy
     @student = Student.of(current_user).find(params[:id])
```
Comment (xingyif) [inline app/controllers/students_controller.rb:139]: already done by `:set_student`
i can delete them in my new pr if these are redundant.

**Comment by zack** [comment]
@xingyif yeah those can all be replaced by adding the method names to the `set_student` array at the top. Good catch!

**Pull request merged**